pkg/providers/tinkerbell/upgrade.go

```diff
@@ -244,7 +245,21 @@ func (p *Provider) ValidateNewSpec(ctx context.Context, cluster *types.Cluster,
 	oSpec := prevDatacenterConfig.Spec
 	nSpec := p.datacenterConfig.Spec
 
-	prevMachineConfigRefs := machineRefSliceToMap(prevSpec.MachineConfigRefs())
+	err = validateControlPlaneMachineConfigRefUnchanged(prevSpec, clusterSpec.Cluster)
```

Comment:
what are you thoughts on the `if err := fn(); err != nil { return err }` pattern?

Reply:
I'd rather stick to 100 chars and extend vertically.